In [1]:
# supress Warnings

import warnings
warnings.filterwarnings('ignore')

import numpy as np
import pandas as pd

In [2]:
# reading the Transaction data

df = pd.read_excel('Transactional Data.xlsx',sheet_name = 'Transaction data (Problem 1)')
df

KO ID  Terminal ID Transaction Date & Time  \
0      DYTXINVGDXXWQUZFKFNWUBK572414037E   20052802.0  28-08-2019 04:57:59 PM   
1      VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0  26-08-2019 10:57:46 AM   
2      FELSPGIAJBRGZTYJDDYNUGS938563554Y   21093301.0  27-08-2019 05:39:11 PM   
3      SITVATJDWOLFBTKKYTIROCR859330456M   19771701.0  28-08-2019 03:58:21 PM   
4      XDRYZQNKPMGKGRSQMUWGUIH454602787D   12834401.0  26-08-2019 03:04:29 PM   
...                                  ...          ...                     ...   
20809  FREVZHFTFTRRGWJPGBCSTLG912170891E   21887901.0  29-08-2019 03:27:30 PM   
20810  NAFODJCVKIPGNTEAETKNORN253103462T   14943001.0  29-08-2019 11:41:47 AM   
20811  AMRVTCHCCVYGDCYJIPHGSMQ270720287N   15807601.0  26-08-2019 02:47:49 PM   
20812  OJORXJHBAAPPWLKETOJBSJQ848706609F   19772401.0  28-08-2019 05:50:08 PM   
20813  FVUNCSANBTRTJWPXCYRQPLO630759211R   10789801.0  26-08-2019 05:40:22 PM   

       Reference Number    Type of Transaction From Account   To Account  \
0      1710764877362961   AEPS ONUS Withdrawal  XXXXXXQMAQA  XXXXXXMFETF   
1      1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
2      1714283137325664   AEPS ONUS Withdrawal  XXXXXXPTMTC  XXXXXXMFETF   
3      1707413971387797   AEPS ONUS Withdrawal  XXXXXXAFMMF  XXXXXXMFETF   
4      1618475279799802      ATM FUNDSTRANSFER  XXXXXXCPGTA  XXXXXXCTCGO   
...                 ...                    ...          ...          ...   
20809  1879173767681724             Withdrawal  XXXXXXTQCFA  XXXXXXMFETF   
20810  1865444735497090   AEPS ONUS Withdrawal  XXXXXXQQEFF  XXXXXXMFETF   
20811  1617396480278196   AEPS ONUS Withdrawal  XXXXXXFGTEC  XXXXXXMFETF   
20812  1801214597177631    ATM Onus Withdrawal  XXXXXXMGTQE  XXXXXXMFETF   
20813  1540735713845884                Deposit  XXXXXXMFETF  XXXXXXFFCFM   

       Amount  Customer Charge  
0        2000              0.0  
1        2500              0.0  
2        2530              0.0  
3        8000              0.0  
4       20000              0.0  
...       ...              ...  
20809    3000              0.0  
20810   10000              0.0  
20811     100              0.0  
20812   10000              0.0  
20813    5000              0.0  

[20814 rows x 9 columns]

#### Split Transactions

In [3]:
# creating a dataset of the split transactions

single = df[['From Account','To Account']].drop_duplicates(keep=False)
split = df[~df.index.isin(single.index)]
split

KO ID  Terminal ID Transaction Date & Time  \
0      DYTXINVGDXXWQUZFKFNWUBK572414037E   20052802.0  28-08-2019 04:57:59 PM   
1      VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0  26-08-2019 10:57:46 AM   
3      SITVATJDWOLFBTKKYTIROCR859330456M   19771701.0  28-08-2019 03:58:21 PM   
6      ZIGIGBOWXHTLEBDQJKRWWPY554861396T   16381401.0  29-08-2019 12:17:55 PM   
7      KEZAQHGDOECFLOUHQPZXUYP724000057H   19460101.0  26-08-2019 03:31:36 PM   
...                                  ...          ...                     ...   
20803  FVUNCSANBTRTJWPXCYRQPLO630759211R   10789801.0  27-08-2019 04:52:45 PM   
20805  SHFWHHXWJUDPSLMTSHBNCIQ463832613H   20033001.0  29-08-2019 03:04:41 PM   
20809  FREVZHFTFTRRGWJPGBCSTLG912170891E   21887901.0  29-08-2019 03:27:30 PM   
20811  AMRVTCHCCVYGDCYJIPHGSMQ270720287N   15807601.0  26-08-2019 02:47:49 PM   
20813  FVUNCSANBTRTJWPXCYRQPLO630759211R   10789801.0  26-08-2019 05:40:22 PM   

       Reference Number    Type of Transaction From Account   To Account  \
0      1710764877362961   AEPS ONUS Withdrawal  XXXXXXQMAQA  XXXXXXMFETF   
1      1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
3      1707413971387797   AEPS ONUS Withdrawal  XXXXXXAFMMF  XXXXXXMFETF   
6      1780565750489285             Withdrawal  XXXXXXAAFOQ  XXXXXXMFETF   
7      1533022971972446   ATM Offus Withdrawal  XXXXXXPCOMP  XXXXXXMFETF   
...                 ...                    ...          ...          ...   
20803  1711514516424200                Deposit  XXXXXXMFETF  XXXXXXTEPOT   
20805  1877831868631329   AEPS ONUS Withdrawal  XXXXXXMPTMC  XXXXXXMFETF   
20809  1879173767681724             Withdrawal  XXXXXXTQCFA  XXXXXXMFETF   
20811  1617396480278196   AEPS ONUS Withdrawal  XXXXXXFGTEC  XXXXXXMFETF   
20813  1540735713845884                Deposit  XXXXXXMFETF  XXXXXXFFCFM   

       Amount  Customer Charge  
0        2000              0.0  
1        2500              0.0  
3        8000              0.0  
6        2900              0.0  
7       10000              0.0  
...       ...              ...  
20803    2000              0.0  
20805    5000              0.0  
20809    3000              0.0  
20811     100              0.0  
20813    5000              0.0  

[12123 rows x 9 columns]

In [4]:
# reading the transaction limits

limit = pd.read_excel('Transactional Data.xlsx',sheet_name = 'Transaction Limit')
limit.rename(columns = {'Transaction Type':'Type of Transaction'},inplace = True)
limit

Type of Transaction  Limit
0   AEPS OFFUS Fund Transfer  10000
1      AEPS OFFUS Withdrawal  10000
2          AEPS ONUS Deposit  10000
3    AEPS ONUS Fund Transfer  10000
4       AEPS ONUS Withdrawal  10000
5                ATM DEPOSIT  19900
6          ATM FUNDSTRANSFER  20000
7       ATM Offus Withdrawal  10000
8        ATM Onus Withdrawal  20000
9                    Deposit  20000
10             Fund Transfer  20000
11       IMPS Money Transfer   5000
12           Initial deposit   1000
13              Loan Deposit  20000
14            Money Transfer  20100
15                Withdrawal  20000
16           YONO WITHDRAWAL  10000

In [5]:
# changing the date time format

split['Transaction Date & Time'] = pd.to_datetime(split['Transaction Date & Time'])
split['Transaction Date & Time'] = [d.date() for d in split['Transaction Date & Time']]
split

KO ID  Terminal ID Transaction Date & Time  \
0      DYTXINVGDXXWQUZFKFNWUBK572414037E   20052802.0              2019-08-28   
1      VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0              2019-08-26   
3      SITVATJDWOLFBTKKYTIROCR859330456M   19771701.0              2019-08-28   
6      ZIGIGBOWXHTLEBDQJKRWWPY554861396T   16381401.0              2019-08-29   
7      KEZAQHGDOECFLOUHQPZXUYP724000057H   19460101.0              2019-08-26   
...                                  ...          ...                     ...   
20803  FVUNCSANBTRTJWPXCYRQPLO630759211R   10789801.0              2019-08-27   
20805  SHFWHHXWJUDPSLMTSHBNCIQ463832613H   20033001.0              2019-08-29   
20809  FREVZHFTFTRRGWJPGBCSTLG912170891E   21887901.0              2019-08-29   
20811  AMRVTCHCCVYGDCYJIPHGSMQ270720287N   15807601.0              2019-08-26   
20813  FVUNCSANBTRTJWPXCYRQPLO630759211R   10789801.0              2019-08-26   

       Reference Number    Type of Transaction From Account   To Account  \
0      1710764877362961   AEPS ONUS Withdrawal  XXXXXXQMAQA  XXXXXXMFETF   
1      1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
3      1707413971387797   AEPS ONUS Withdrawal  XXXXXXAFMMF  XXXXXXMFETF   
6      1780565750489285             Withdrawal  XXXXXXAAFOQ  XXXXXXMFETF   
7      1533022971972446   ATM Offus Withdrawal  XXXXXXPCOMP  XXXXXXMFETF   
...                 ...                    ...          ...          ...   
20803  1711514516424200                Deposit  XXXXXXMFETF  XXXXXXTEPOT   
20805  1877831868631329   AEPS ONUS Withdrawal  XXXXXXMPTMC  XXXXXXMFETF   
20809  1879173767681724             Withdrawal  XXXXXXTQCFA  XXXXXXMFETF   
20811  1617396480278196   AEPS ONUS Withdrawal  XXXXXXFGTEC  XXXXXXMFETF   
20813  1540735713845884                Deposit  XXXXXXMFETF  XXXXXXFFCFM   

       Amount  Customer Charge  
0        2000              0.0  
1        2500              0.0  
3        8000              0.0  
6        2900              0.0  
7       10000              0.0  
...       ...              ...  
20803    2000              0.0  
20805    5000              0.0  
20809    3000              0.0  
20811     100              0.0  
20813    5000              0.0  

[12123 rows x 9 columns]

In [6]:
split['KO ID / Type of Transaction'] = split['KO ID'] + ' / ' + split['Type of Transaction']
split.drop(['KO ID','Type of Transaction'],axis = 1,inplace = True)
split

Terminal ID Transaction Date & Time  Reference Number From Account  \
0       20052802.0              2019-08-28  1710764877362961  XXXXXXQMAQA   
1       14017601.0              2019-08-26  1516378236346888  XXXXXXCCOMP   
3       19771701.0              2019-08-28  1707413971387797  XXXXXXAFMMF   
6       16381401.0              2019-08-29  1780565750489285  XXXXXXAAFOQ   
7       19460101.0              2019-08-26  1533022971972446  XXXXXXPCOMP   
...            ...                     ...               ...          ...   
20803   10789801.0              2019-08-27  1711514516424200  XXXXXXMFETF   
20805   20033001.0              2019-08-29  1877831868631329  XXXXXXMPTMC   
20809   21887901.0              2019-08-29  1879173767681724  XXXXXXTQCFA   
20811   15807601.0              2019-08-26  1617396480278196  XXXXXXFGTEC   
20813   10789801.0              2019-08-26  1540735713845884  XXXXXXMFETF   

        To Account  Amount  Customer Charge  \
0      XXXXXXMFETF    2000              0.0   
1      XXXXXXMFETF    2500              0.0   
3      XXXXXXMFETF    8000              0.0   
6      XXXXXXMFETF    2900              0.0   
7      XXXXXXMFETF   10000              0.0   
...            ...     ...              ...   
20803  XXXXXXTEPOT    2000              0.0   
20805  XXXXXXMFETF    5000              0.0   
20809  XXXXXXMFETF    3000              0.0   
20811  XXXXXXMFETF     100              0.0   
20813  XXXXXXFFCFM    5000              0.0   

                             KO ID / Type of Transaction  
0      DYTXINVGDXXWQUZFKFNWUBK572414037E / AEPS ONUS ...  
1      VHONSNSYIIPJYCPGUKKGBKY795411271J / AEPS OFFUS...  
3      SITVATJDWOLFBTKKYTIROCR859330456M / AEPS ONUS ...  
6         ZIGIGBOWXHTLEBDQJKRWWPY554861396T / Withdrawal  
7      KEZAQHGDOECFLOUHQPZXUYP724000057H / ATM Offus ...  
...                                                  ...  
20803        FVUNCSANBTRTJWPXCYRQPLO630759211R / Deposit  
20805  SHFWHHXWJUDPSLMTSHBNCIQ463832613H / AEPS ONUS ...  
20809     FREVZHFTFTRRGWJPGBCSTLG912170891E / Withdrawal  
20811  AMRVTCHCCVYGDCYJIPHGSMQ270720287N / AEPS ONUS ...  
20813        FVUNCSANBTRTJWPXCYRQPLO630759211R / Deposit  

[12123 rows x 8 columns]

In [7]:
# summing the amount of split transactions

df1 = split.groupby(by = ['KO ID / Type of Transaction','Transaction Date & Time','From Account','To Account'])['Amount'].sum().reset_index()
df1

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
1     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
2     AAETIGMJYSSPYTYXPASJWSD580547108H / AEPS ONUS ...   
3        AAETIGMJYSSPYTYXPASJWSD580547108H / Withdrawal   
4        AAETIGMJYSSPYTYXPASJWSD580547108H / Withdrawal   
...                                                 ...   
7146  ZZONEJSIFVIVDVWDEPIGSIE629385895J / AEPS ONUS ...   
7147  ZZONEJSIFVIVDVWDEPIGSIE629385895J / AEPS ONUS ...   
7148  ZZONEJSIFVIVDVWDEPIGSIE629385895J / AEPS ONUS ...   
7149  ZZONEJSIFVIVDVWDEPIGSIE629385895J / ATM Offus ...   
7150  ZZONEJSIFVIVDVWDEPIGSIE629385895J / Money Tran...   

     Transaction Date & Time From Account   To Account  Amount  
0                 2019-08-28  XXXXXXOPOMQ  XXXXXXMFETF   10000  
1                 2019-08-29  XXXXXXOPOMQ  XXXXXXMFETF    5000  
2                 2019-08-27  XXXXXXMFETF  XXXXXXCMTPP    2500  
3                 2019-08-26  XXXXXXQAQQM  XXXXXXMFETF    7500  
4                 2019-08-26  XXXXXXQPCFC  XXXXXXMFETF    4000  
...                      ...          ...          ...     ...  
7146              2019-08-26  XXXXXXPMOGQ  XXXXXXMFETF    2000  
7147              2019-08-27  XXXXXXPMOGQ  XXXXXXMFETF    5500  
7148              2019-08-29  XXXXXXTETTQ  XXXXXXMFETF    7000  
7149              2019-08-26  XXXXXXPCOMP  XXXXXXMFETF     500  
7150              2019-08-26  XXXXXXMFETF  XXXXXXFFETP    1000  

[7151 rows x 5 columns]

In [8]:
df1['Type of Transaction'] = [d.split(' / ')[1] for d in df1['KO ID / Type of Transaction']]
df1

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
1     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
2     AAETIGMJYSSPYTYXPASJWSD580547108H / AEPS ONUS ...   
3        AAETIGMJYSSPYTYXPASJWSD580547108H / Withdrawal   
4        AAETIGMJYSSPYTYXPASJWSD580547108H / Withdrawal   
...                                                 ...   
7146  ZZONEJSIFVIVDVWDEPIGSIE629385895J / AEPS ONUS ...   
7147  ZZONEJSIFVIVDVWDEPIGSIE629385895J / AEPS ONUS ...   
7148  ZZONEJSIFVIVDVWDEPIGSIE629385895J / AEPS ONUS ...   
7149  ZZONEJSIFVIVDVWDEPIGSIE629385895J / ATM Offus ...   
7150  ZZONEJSIFVIVDVWDEPIGSIE629385895J / Money Tran...   

     Transaction Date & Time From Account   To Account  Amount  \
0                 2019-08-28  XXXXXXOPOMQ  XXXXXXMFETF   10000   
1                 2019-08-29  XXXXXXOPOMQ  XXXXXXMFETF    5000   
2                 2019-08-27  XXXXXXMFETF  XXXXXXCMTPP    2500   
3                 2019-08-26  XXXXXXQAQQM  XXXXXXMFETF    7500   
4                 2019-08-26  XXXXXXQPCFC  XXXXXXMFETF    4000   
...                      ...          ...          ...     ...   
7146              2019-08-26  XXXXXXPMOGQ  XXXXXXMFETF    2000   
7147              2019-08-27  XXXXXXPMOGQ  XXXXXXMFETF    5500   
7148              2019-08-29  XXXXXXTETTQ  XXXXXXMFETF    7000   
7149              2019-08-26  XXXXXXPCOMP  XXXXXXMFETF     500   
7150              2019-08-26  XXXXXXMFETF  XXXXXXFFETP    1000   

       Type of Transaction  
0     AEPS ONUS Withdrawal  
1     AEPS ONUS Withdrawal  
2        AEPS ONUS Deposit  
3               Withdrawal  
4               Withdrawal  
...                    ...  
7146  AEPS ONUS Withdrawal  
7147  AEPS ONUS Withdrawal  
7148  AEPS ONUS Withdrawal  
7149  ATM Offus Withdrawal  
7150        Money Transfer  

[7151 rows x 6 columns]

In [9]:
# merging the limits

merged = pd.merge(df1,limit,how = 'inner',on = 'Type of Transaction')
merged.drop('Type of Transaction',axis = 1,inplace = True)
merged

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
1     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
2     AAGOFWQTKISCPXQXBIFKAAC509419983N / AEPS ONUS ...   
3     AAMIHDZNHEQEDPJNONUMGHB649383828V / AEPS ONUS ...   
4     AAMIHDZNHEQEDPJNONUMGHB649383828V / AEPS ONUS ...   
...                                                 ...   
7144   KSNETQPNXFHPBNCGSTYSQNG444666297D / Loan Deposit   
7145   VMYGADUZKYAUPYPCIIAHQMD495329355Y / Loan Deposit   
7146   VTSAMKFAQRIGNJEWCFCMJXZ945886627K / Loan Deposit   
7147     LZFXISCDPREWAVFNLQJQHBH82188323G / ATM DEPOSIT   
7148  ORLJLSOWWNGYQLVKIXNCHFT37145667T / ATM FUNDSTR...   

     Transaction Date & Time From Account   To Account  Amount  Limit  
0                 2019-08-28  XXXXXXOPOMQ  XXXXXXMFETF   10000  10000  
1                 2019-08-29  XXXXXXOPOMQ  XXXXXXMFETF    5000  10000  
2                 2019-08-28  XXXXXXCOGMM  XXXXXXMFETF   10000  10000  
3                 2019-08-26  XXXXXXCMAAO  XXXXXXMFETF    1000  10000  
4                 2019-08-29  XXXXXXFFTFT  XXXXXXMFETF     500  10000  
...                      ...          ...          ...     ...    ...  
7144              2019-08-26  XXXXXXMFETF  XXXXXXMAETQ   15500  20000  
7145              2019-08-29  XXXXXXMFETF  XXXXXXTTPOP    5000  20000  
7146              2019-08-26  XXXXXXMFETF  XXXXXXAQGCP   20000  20000  
7147              2019-08-27  XXXXXXMFETF  XXXXXXPAGFA   19900  19900  
7148              2019-08-28  XXXXXXCMAPP  XXXXXXQGPTP    2400  20000  

[7149 rows x 6 columns]

In [10]:
# those split transactions whose total amount is less than or equal to the limit of transaction
# are suspicious transactions

suspicious_split_transactions = merged[merged['Amount']<=merged['Limit']]
suspicious_split_transactions

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
1     AAAORAUTPSCQXOGVNELKNCB187842400C / AEPS ONUS ...   
2     AAGOFWQTKISCPXQXBIFKAAC509419983N / AEPS ONUS ...   
3     AAMIHDZNHEQEDPJNONUMGHB649383828V / AEPS ONUS ...   
4     AAMIHDZNHEQEDPJNONUMGHB649383828V / AEPS ONUS ...   
...                                                 ...   
7144   KSNETQPNXFHPBNCGSTYSQNG444666297D / Loan Deposit   
7145   VMYGADUZKYAUPYPCIIAHQMD495329355Y / Loan Deposit   
7146   VTSAMKFAQRIGNJEWCFCMJXZ945886627K / Loan Deposit   
7147     LZFXISCDPREWAVFNLQJQHBH82188323G / ATM DEPOSIT   
7148  ORLJLSOWWNGYQLVKIXNCHFT37145667T / ATM FUNDSTR...   

     Transaction Date & Time From Account   To Account  Amount  Limit  
0                 2019-08-28  XXXXXXOPOMQ  XXXXXXMFETF   10000  10000  
1                 2019-08-29  XXXXXXOPOMQ  XXXXXXMFETF    5000  10000  
2                 2019-08-28  XXXXXXCOGMM  XXXXXXMFETF   10000  10000  
3                 2019-08-26  XXXXXXCMAAO  XXXXXXMFETF    1000  10000  
4                 2019-08-29  XXXXXXFFTFT  XXXXXXMFETF     500  10000  
...                      ...          ...          ...     ...    ...  
7144              2019-08-26  XXXXXXMFETF  XXXXXXMAETQ   15500  20000  
7145              2019-08-29  XXXXXXMFETF  XXXXXXTTPOP    5000  20000  
7146              2019-08-26  XXXXXXMFETF  XXXXXXAQGCP   20000  20000  
7147              2019-08-27  XXXXXXMFETF  XXXXXXPAGFA   19900  19900  
7148              2019-08-28  XXXXXXCMAPP  XXXXXXQGPTP    2400  20000  

[7136 rows x 6 columns]

In [11]:
# creating an excel file of the suspicious split transactions

suspicious_split_transactions.to_excel(r'Suspicious Split Transactions.xlsx',index = False)

#### Round Trip Transactions

In [12]:
df

KO ID  Terminal ID Transaction Date & Time  \
0      DYTXINVGDXXWQUZFKFNWUBK572414037E   20052802.0  28-08-2019 04:57:59 PM   
1      VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0  26-08-2019 10:57:46 AM   
2      FELSPGIAJBRGZTYJDDYNUGS938563554Y   21093301.0  27-08-2019 05:39:11 PM   
3      SITVATJDWOLFBTKKYTIROCR859330456M   19771701.0  28-08-2019 03:58:21 PM   
4      XDRYZQNKPMGKGRSQMUWGUIH454602787D   12834401.0  26-08-2019 03:04:29 PM   
...                                  ...          ...                     ...   
20809  FREVZHFTFTRRGWJPGBCSTLG912170891E   21887901.0  29-08-2019 03:27:30 PM   
20810  NAFODJCVKIPGNTEAETKNORN253103462T   14943001.0  29-08-2019 11:41:47 AM   
20811  AMRVTCHCCVYGDCYJIPHGSMQ270720287N   15807601.0  26-08-2019 02:47:49 PM   
20812  OJORXJHBAAPPWLKETOJBSJQ848706609F   19772401.0  28-08-2019 05:50:08 PM   
20813  FVUNCSANBTRTJWPXCYRQPLO630759211R   10789801.0  26-08-2019 05:40:22 PM   

       Reference Number    Type of Transaction From Account   To Account  \
0      1710764877362961   AEPS ONUS Withdrawal  XXXXXXQMAQA  XXXXXXMFETF   
1      1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
2      1714283137325664   AEPS ONUS Withdrawal  XXXXXXPTMTC  XXXXXXMFETF   
3      1707413971387797   AEPS ONUS Withdrawal  XXXXXXAFMMF  XXXXXXMFETF   
4      1618475279799802      ATM FUNDSTRANSFER  XXXXXXCPGTA  XXXXXXCTCGO   
...                 ...                    ...          ...          ...   
20809  1879173767681724             Withdrawal  XXXXXXTQCFA  XXXXXXMFETF   
20810  1865444735497090   AEPS ONUS Withdrawal  XXXXXXQQEFF  XXXXXXMFETF   
20811  1617396480278196   AEPS ONUS Withdrawal  XXXXXXFGTEC  XXXXXXMFETF   
20812  1801214597177631    ATM Onus Withdrawal  XXXXXXMGTQE  XXXXXXMFETF   
20813  1540735713845884                Deposit  XXXXXXMFETF  XXXXXXFFCFM   

       Amount  Customer Charge  
0        2000              0.0  
1        2500              0.0  
2        2530              0.0  
3        8000              0.0  
4       20000              0.0  
...       ...              ...  
20809    3000              0.0  
20810   10000              0.0  
20811     100              0.0  
20812   10000              0.0  
20813    5000              0.0  

[20814 rows x 9 columns]

In [13]:
from_account = df['From Account'].value_counts().reset_index()
from_account.rename(columns = {'index':'From Account','From Account':'Count'},inplace = True)
from_account

From Account  Count
0     XXXXXXMFETF   5359
1     XXXXXXPCOMP    245
2     XXXXXXCCOMP    180
3     XXXXXXACOMP    149
4     XXXXXXECOMP    131
...           ...    ...
9290  XXXXXXAAOMA      1
9291  XXXXXXOMGEA      1
9292  XXXXXXOMPCG      1
9293  XXXXXXOFCAF      1
9294  XXXXXXTGGEE      1

[9295 rows x 2 columns]

In [14]:
to_account = df['To Account'].value_counts().reset_index()
to_account.rename(columns = {'index':'To Account','To Account':'Count'},inplace = True)
to_account

To Account  Count
0     XXXXXXMFETF  13814
1     XXXXXXCGOOE    128
2     XXXXXXMAGGA     96
3     XXXXXXTGPMT     56
4     XXXXXXQPFME     42
...           ...    ...
4666  XXXXXXFFCMT      1
4667  XXXXXXTFCCM      1
4668  XXXXXXQPGGM      1
4669  XXXXXXAEAGP      1
4670  XXXXXXMFEFC      1

[4671 rows x 2 columns]

Clearly there are some round trip transactions happening.

In [15]:
# we can identify the number of transactions happening by merging the two dataframes on the accounts

to_account.rename(columns = {'To Account':'From Account'},inplace = True)
to_account

From Account  Count
0     XXXXXXMFETF  13814
1     XXXXXXCGOOE    128
2     XXXXXXMAGGA     96
3     XXXXXXTGPMT     56
4     XXXXXXQPFME     42
...           ...    ...
4666  XXXXXXFFCMT      1
4667  XXXXXXTFCCM      1
4668  XXXXXXQPGGM      1
4669  XXXXXXAEAGP      1
4670  XXXXXXMFEFC      1

[4671 rows x 2 columns]

In [16]:
merged = pd.merge(from_account,to_account,how = 'inner',on = 'From Account')
merged

From Account  Count_x  Count_y
0    XXXXXXMFETF     5359    13814
1    XXXXXXPCOMP      245        1
2    XXXXXXCCOMP      180        1
3    XXXXXXACOMP      149        2
4    XXXXXXECOMP      131        3
..           ...      ...      ...
586  XXXXXXACAEF        1        1
587  XXXXXXMFEFC        1        1
588  XXXXXXAGAGF        1        1
589  XXXXXXTOOCA        1        1
590  XXXXXXOMPCG        1        1

[591 rows x 3 columns]

There are 591 such accounts which have round trip transactions going on.

In [17]:
round_trip_transactions = pd.merge(df,merged,how = 'inner',on = 'From Account')
round_trip_transactions

KO ID  Terminal ID Transaction Date & Time  \
0     VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0  26-08-2019 10:57:46 AM   
1     VZRQMMEVHBCVTSUTNOFGYUU779875018C   21527101.0  26-08-2019 11:22:56 AM   
2     LROIJLLWQYFCPPDSLULUIPV234358262A   20924501.0  26-08-2019 07:15:50 PM   
3     JNUFMAJCWATETOREQKBVBDL575031176I   21471801.0  26-08-2019 10:21:22 AM   
4     TKDSATSKIFLVYBFTRWNTQGU359506718R   16682101.0  26-08-2019 06:41:53 PM   
...                                 ...          ...                     ...   
7193  OFGBXILCGBOSLQZKTQXRINN622024505T   20977801.0  29-08-2019 03:45:54 PM   
7194   BAHGRTGEVIMQBJCCQUCVPCU44126938J   20378701.0  29-08-2019 01:01:22 PM   
7195  WRWIEODLKKSZIFBSNXEYSIW556129742M   18055901.0  26-08-2019 08:59:08 AM   
7196  KSNETQPNXFHPBNCGSTYSQNG444666297D   12793201.0  27-08-2019 08:53:28 AM   
7197  XXKNLIXAIHENMAHRLQVOPIA533902231X   10749101.0  27-08-2019 06:51:00 PM   

      Reference Number    Type of Transaction From Account   To Account  \
0     1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
1     1605167480615215  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
2     1633491250740161  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
3     1601445393627309  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
4     1544108972528877  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
...                ...                    ...          ...          ...   
7193  1880101597877850   AEPS ONUS Withdrawal  XXXXXXAOCGT  XXXXXXMFETF   
7194  1782878457178134   AEPS ONUS Withdrawal  XXXXXXFOQEE  XXXXXXMFETF   
7195  1509247652449944         Money Transfer  XXXXXXOOFCM  XXXXXXMQMFC   
7196  1682614421929108    ATM Onus Withdrawal  XXXXXXAQAEO  XXXXXXMFETF   
7197  1631050654115233   AEPS ONUS Withdrawal  XXXXXXAOCMP  XXXXXXMFETF   

      Amount  Customer Charge  Count_x  Count_y  
0       2500              0.0      180        1  
1       2000              0.0      180        1  
2        100              0.0      180        1  
3       2200              0.0      180        1  
4       2000              0.0      180        1  
...      ...              ...      ...      ...  
7193    2000              0.0        1        1  
7194    1000              0.0        1        3  
7195    2000              0.0        1        3  
7196    3000              0.0        1        1  
7197    1100              0.0        1        1  

[7198 rows x 11 columns]

There are 7198 such round trip transactions.

In [18]:
# creating an excel file of the suspicious round trip transactions

round_trip_transactions.to_excel(r'Suspicious Round Trip Transactions.xlsx',index = False)

#### Round + Split Transactions

In [19]:
round_trip_transactions

KO ID  Terminal ID Transaction Date & Time  \
0     VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0  26-08-2019 10:57:46 AM   
1     VZRQMMEVHBCVTSUTNOFGYUU779875018C   21527101.0  26-08-2019 11:22:56 AM   
2     LROIJLLWQYFCPPDSLULUIPV234358262A   20924501.0  26-08-2019 07:15:50 PM   
3     JNUFMAJCWATETOREQKBVBDL575031176I   21471801.0  26-08-2019 10:21:22 AM   
4     TKDSATSKIFLVYBFTRWNTQGU359506718R   16682101.0  26-08-2019 06:41:53 PM   
...                                 ...          ...                     ...   
7193  OFGBXILCGBOSLQZKTQXRINN622024505T   20977801.0  29-08-2019 03:45:54 PM   
7194   BAHGRTGEVIMQBJCCQUCVPCU44126938J   20378701.0  29-08-2019 01:01:22 PM   
7195  WRWIEODLKKSZIFBSNXEYSIW556129742M   18055901.0  26-08-2019 08:59:08 AM   
7196  KSNETQPNXFHPBNCGSTYSQNG444666297D   12793201.0  27-08-2019 08:53:28 AM   
7197  XXKNLIXAIHENMAHRLQVOPIA533902231X   10749101.0  27-08-2019 06:51:00 PM   

      Reference Number    Type of Transaction From Account   To Account  \
0     1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
1     1605167480615215  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
2     1633491250740161  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
3     1601445393627309  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
4     1544108972528877  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
...                ...                    ...          ...          ...   
7193  1880101597877850   AEPS ONUS Withdrawal  XXXXXXAOCGT  XXXXXXMFETF   
7194  1782878457178134   AEPS ONUS Withdrawal  XXXXXXFOQEE  XXXXXXMFETF   
7195  1509247652449944         Money Transfer  XXXXXXOOFCM  XXXXXXMQMFC   
7196  1682614421929108    ATM Onus Withdrawal  XXXXXXAQAEO  XXXXXXMFETF   
7197  1631050654115233   AEPS ONUS Withdrawal  XXXXXXAOCMP  XXXXXXMFETF   

      Amount  Customer Charge  Count_x  Count_y  
0       2500              0.0      180        1  
1       2000              0.0      180        1  
2        100              0.0      180        1  
3       2200              0.0      180        1  
4       2000              0.0      180        1  
...      ...              ...      ...      ...  
7193    2000              0.0        1        1  
7194    1000              0.0        1        3  
7195    2000              0.0        1        3  
7196    3000              0.0        1        1  
7197    1100              0.0        1        1  

[7198 rows x 11 columns]

In [20]:
# creating a dataset of the round + split transactions

single1 = round_trip_transactions[['From Account','To Account']].drop_duplicates(keep=False)
split1 = round_trip_transactions[~round_trip_transactions.index.isin(single.index)]
split1

KO ID  Terminal ID Transaction Date & Time  \
0     VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0  26-08-2019 10:57:46 AM   
1     VZRQMMEVHBCVTSUTNOFGYUU779875018C   21527101.0  26-08-2019 11:22:56 AM   
3     JNUFMAJCWATETOREQKBVBDL575031176I   21471801.0  26-08-2019 10:21:22 AM   
6     MXNFVTDNHQXHQOXQKJFCIIL303070411Q   20586201.0  26-08-2019 11:57:05 AM   
7     ODMMEQUSZBUOMJTAQQEZITV683070080I   21443301.0  26-08-2019 06:15:59 PM   
...                                 ...          ...                     ...   
7190  DOVDUYJXYABEGBILJLUXLDM759008830O   10493902.0  27-08-2019 09:27:25 AM   
7191  OYXXHGVXFJZLFAQRGURLAFC893874109B   10789601.0  26-08-2019 10:29:55 AM   
7192  ARTMNKJYPAXUSDXLGJHUQZP552079700P   17512101.0  26-08-2019 12:56:18 PM   
7193  OFGBXILCGBOSLQZKTQXRINN622024505T   20977801.0  29-08-2019 03:45:54 PM   
7197  XXKNLIXAIHENMAHRLQVOPIA533902231X   10749101.0  27-08-2019 06:51:00 PM   

       Reference Number    Type of Transaction From Account   To Account  \
0      1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
1      1605167480615215  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
3      1601445393627309  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
6      1520134634071787  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
7      1542868184063944  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
...                 ...                    ...          ...          ...   
7190   1597213761149623             Withdrawal  XXXXXXAPACQ  XXXXXXMFETF   
7191  12972169451149624             Withdrawal  XXXXXXOMPCG  XXXXXXMFETF   
7192   1523672816670929             Withdrawal  XXXXXXCQEAQ  XXXXXXMFETF   
7193   1880101597877850   AEPS ONUS Withdrawal  XXXXXXAOCGT  XXXXXXMFETF   
7197   1631050654115233   AEPS ONUS Withdrawal  XXXXXXAOCMP  XXXXXXMFETF   

      Amount  Customer Charge  Count_x  Count_y  
0       2500              0.0      180        1  
1       2000              0.0      180        1  
3       2200              0.0      180        1  
6        400              0.0      180        1  
7        500              0.0      180        1  
...      ...              ...      ...      ...  
7190    6000              0.0        1        1  
7191   10000              0.0        1        1  
7192    2000              0.0        1        1  
7193    2000              0.0        1        1  
7197    1100              0.0        1        1  

[4216 rows x 11 columns]

In [21]:
# changing the date time format

split1['Transaction Date & Time'] = pd.to_datetime(split1['Transaction Date & Time'])
split1['Transaction Date & Time'] = [d.date() for d in split1['Transaction Date & Time']]
split1

KO ID  Terminal ID Transaction Date & Time  \
0     VHONSNSYIIPJYCPGUKKGBKY795411271J   14017601.0              2019-08-26   
1     VZRQMMEVHBCVTSUTNOFGYUU779875018C   21527101.0              2019-08-26   
3     JNUFMAJCWATETOREQKBVBDL575031176I   21471801.0              2019-08-26   
6     MXNFVTDNHQXHQOXQKJFCIIL303070411Q   20586201.0              2019-08-26   
7     ODMMEQUSZBUOMJTAQQEZITV683070080I   21443301.0              2019-08-26   
...                                 ...          ...                     ...   
7190  DOVDUYJXYABEGBILJLUXLDM759008830O   10493902.0              2019-08-27   
7191  OYXXHGVXFJZLFAQRGURLAFC893874109B   10789601.0              2019-08-26   
7192  ARTMNKJYPAXUSDXLGJHUQZP552079700P   17512101.0              2019-08-26   
7193  OFGBXILCGBOSLQZKTQXRINN622024505T   20977801.0              2019-08-29   
7197  XXKNLIXAIHENMAHRLQVOPIA533902231X   10749101.0              2019-08-27   

       Reference Number    Type of Transaction From Account   To Account  \
0      1516378236346888  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
1      1605167480615215  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
3      1601445393627309  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
6      1520134634071787  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
7      1542868184063944  AEPS OFFUS Withdrawal  XXXXXXCCOMP  XXXXXXMFETF   
...                 ...                    ...          ...          ...   
7190   1597213761149623             Withdrawal  XXXXXXAPACQ  XXXXXXMFETF   
7191  12972169451149624             Withdrawal  XXXXXXOMPCG  XXXXXXMFETF   
7192   1523672816670929             Withdrawal  XXXXXXCQEAQ  XXXXXXMFETF   
7193   1880101597877850   AEPS ONUS Withdrawal  XXXXXXAOCGT  XXXXXXMFETF   
7197   1631050654115233   AEPS ONUS Withdrawal  XXXXXXAOCMP  XXXXXXMFETF   

      Amount  Customer Charge  Count_x  Count_y  
0       2500              0.0      180        1  
1       2000              0.0      180        1  
3       2200              0.0      180        1  
6        400              0.0      180        1  
7        500              0.0      180        1  
...      ...              ...      ...      ...  
7190    6000              0.0        1        1  
7191   10000              0.0        1        1  
7192    2000              0.0        1        1  
7193    2000              0.0        1        1  
7197    1100              0.0        1        1  

[4216 rows x 11 columns]

In [22]:
split1['KO ID / Type of Transaction'] = split1['KO ID'] + ' / ' + split1['Type of Transaction']
split1.drop(['KO ID','Type of Transaction'],axis = 1,inplace = True)
split1

Terminal ID Transaction Date & Time   Reference Number From Account  \
0      14017601.0              2019-08-26   1516378236346888  XXXXXXCCOMP   
1      21527101.0              2019-08-26   1605167480615215  XXXXXXCCOMP   
3      21471801.0              2019-08-26   1601445393627309  XXXXXXCCOMP   
6      20586201.0              2019-08-26   1520134634071787  XXXXXXCCOMP   
7      21443301.0              2019-08-26   1542868184063944  XXXXXXCCOMP   
...           ...                     ...                ...          ...   
7190   10493902.0              2019-08-27   1597213761149623  XXXXXXAPACQ   
7191   10789601.0              2019-08-26  12972169451149624  XXXXXXOMPCG   
7192   17512101.0              2019-08-26   1523672816670929  XXXXXXCQEAQ   
7193   20977801.0              2019-08-29   1880101597877850  XXXXXXAOCGT   
7197   10749101.0              2019-08-27   1631050654115233  XXXXXXAOCMP   

       To Account  Amount  Customer Charge  Count_x  Count_y  \
0     XXXXXXMFETF    2500              0.0      180        1   
1     XXXXXXMFETF    2000              0.0      180        1   
3     XXXXXXMFETF    2200              0.0      180        1   
6     XXXXXXMFETF     400              0.0      180        1   
7     XXXXXXMFETF     500              0.0      180        1   
...           ...     ...              ...      ...      ...   
7190  XXXXXXMFETF    6000              0.0        1        1   
7191  XXXXXXMFETF   10000              0.0        1        1   
7192  XXXXXXMFETF    2000              0.0        1        1   
7193  XXXXXXMFETF    2000              0.0        1        1   
7197  XXXXXXMFETF    1100              0.0        1        1   

                            KO ID / Type of Transaction  
0     VHONSNSYIIPJYCPGUKKGBKY795411271J / AEPS OFFUS...  
1     VZRQMMEVHBCVTSUTNOFGYUU779875018C / AEPS OFFUS...  
3     JNUFMAJCWATETOREQKBVBDL575031176I / AEPS OFFUS...  
6     MXNFVTDNHQXHQOXQKJFCIIL303070411Q / AEPS OFFUS...  
7     ODMMEQUSZBUOMJTAQQEZITV683070080I / AEPS OFFUS...  
...                                                 ...  
7190     DOVDUYJXYABEGBILJLUXLDM759008830O / Withdrawal  
7191     OYXXHGVXFJZLFAQRGURLAFC893874109B / Withdrawal  
7192     ARTMNKJYPAXUSDXLGJHUQZP552079700P / Withdrawal  
7193  OFGBXILCGBOSLQZKTQXRINN622024505T / AEPS ONUS ...  
7197  XXKNLIXAIHENMAHRLQVOPIA533902231X / AEPS ONUS ...  

[4216 rows x 10 columns]

In [23]:
# summing the amount of split transactions

df2 = split1.groupby(by = ['KO ID / Type of Transaction','Transaction Date & Time','From Account','To Account'])['Amount'].sum().reset_index()
df2

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / Money Tran...   
1     AAETIGMJYSSPYTYXPASJWSD580547108H / AEPS ONUS ...   
2           AAETIGMJYSSPYTYXPASJWSD580547108H / Deposit   
3     AAETIGMJYSSPYTYXPASJWSD580547108H / Money Tran...   
4     AAMIHDZNHEQEDPJNONUMGHB649383828V / AEPS ONUS ...   
...                                                 ...   
3735  ZYNLDAXVGREWKDHISWRNETR399412215M / Initial de...   
3736  ZZONEJSIFVIVDVWDEPIGSIE629385895J / ATM Offus ...   
3737  ZZONEJSIFVIVDVWDEPIGSIE629385895J / Money Tran...   
3738  ZZONEJSIFVIVDVWDEPIGSIE629385895J / Money Tran...   
3739        ZZTIVUBYXBJSQAUDZSHAVLJ709965574Q / Deposit   

     Transaction Date & Time From Account   To Account  Amount  
0                 2019-08-29  XXXXXXMFETF  XXXXXXEFMEG    3663  
1                 2019-08-27  XXXXXXMFETF  XXXXXXCMTPP    2500  
2                 2019-08-26  XXXXXXMFETF  XXXXXXAQQEP   11000  
3                 2019-08-27  XXXXXXMFETF  XXXXXXTAGTE    2000  
4                 2019-08-26  XXXXXXMFETF  XXXXXXMMTPF   10000  
...                      ...          ...          ...     ...  
3735              2019-08-28  XXXXXXMFETF  XXXXXXFCTQE      50  
3736              2019-08-26  XXXXXXPCOMP  XXXXXXMFETF     500  
3737              2019-08-26  XXXXXXMFETF  XXXXXXFFETP    1000  
3738              2019-08-28  XXXXXXMFETF  XXXXXXPACGO    3960  
3739              2019-08-29  XXXXXXMFETF  XXXXXXQTGCC     100  

[3740 rows x 5 columns]

In [24]:
df2['Type of Transaction'] = [d.split(' / ')[1] for d in df2['KO ID / Type of Transaction']]
df2

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / Money Tran...   
1     AAETIGMJYSSPYTYXPASJWSD580547108H / AEPS ONUS ...   
2           AAETIGMJYSSPYTYXPASJWSD580547108H / Deposit   
3     AAETIGMJYSSPYTYXPASJWSD580547108H / Money Tran...   
4     AAMIHDZNHEQEDPJNONUMGHB649383828V / AEPS ONUS ...   
...                                                 ...   
3735  ZYNLDAXVGREWKDHISWRNETR399412215M / Initial de...   
3736  ZZONEJSIFVIVDVWDEPIGSIE629385895J / ATM Offus ...   
3737  ZZONEJSIFVIVDVWDEPIGSIE629385895J / Money Tran...   
3738  ZZONEJSIFVIVDVWDEPIGSIE629385895J / Money Tran...   
3739        ZZTIVUBYXBJSQAUDZSHAVLJ709965574Q / Deposit   

     Transaction Date & Time From Account   To Account  Amount  \
0                 2019-08-29  XXXXXXMFETF  XXXXXXEFMEG    3663   
1                 2019-08-27  XXXXXXMFETF  XXXXXXCMTPP    2500   
2                 2019-08-26  XXXXXXMFETF  XXXXXXAQQEP   11000   
3                 2019-08-27  XXXXXXMFETF  XXXXXXTAGTE    2000   
4                 2019-08-26  XXXXXXMFETF  XXXXXXMMTPF   10000   
...                      ...          ...          ...     ...   
3735              2019-08-28  XXXXXXMFETF  XXXXXXFCTQE      50   
3736              2019-08-26  XXXXXXPCOMP  XXXXXXMFETF     500   
3737              2019-08-26  XXXXXXMFETF  XXXXXXFFETP    1000   
3738              2019-08-28  XXXXXXMFETF  XXXXXXPACGO    3960   
3739              2019-08-29  XXXXXXMFETF  XXXXXXQTGCC     100   

       Type of Transaction  
0           Money Transfer  
1        AEPS ONUS Deposit  
2                  Deposit  
3           Money Transfer  
4        AEPS ONUS Deposit  
...                    ...  
3735       Initial deposit  
3736  ATM Offus Withdrawal  
3737        Money Transfer  
3738        Money Transfer  
3739               Deposit  

[3740 rows x 6 columns]

In [25]:
# merging the limits

merged = pd.merge(df2,limit,how = 'inner',on = 'Type of Transaction')
merged.drop('Type of Transaction',axis = 1,inplace = True)
merged

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / Money Tran...   
1     AAETIGMJYSSPYTYXPASJWSD580547108H / Money Tran...   
2     AAMIHDZNHEQEDPJNONUMGHB649383828V / Money Tran...   
3     AAPGYETNHRFYWNARXFDRXWJ710966481L / Money Tran...   
4     ABIURLGMYPUCODQTHLKKGOR654931422Z / Money Tran...   
...                                                 ...   
3729  ZVPETZNJGMLVCLFUWMXXCMB565230402U / Fund Transfer   
3730  FYKWLNFXAVGGSPCQVMYMFUG701624369F / AEPS ONUS ...   
3731  IVWTTRJEVPHZCLHUORTVKQR937821643B / AEPS ONUS ...   
3732  HQOEVKFLNTBGNCHLVWFOAMC261533838U / YONO WITHD...   
3733  UUZRCLIQCEKVTCBJHQJSJPS569050463H / AEPS OFFUS...   

     Transaction Date & Time From Account   To Account  Amount  Limit  
0                 2019-08-29  XXXXXXMFETF  XXXXXXEFMEG    3663  20100  
1                 2019-08-27  XXXXXXMFETF  XXXXXXTAGTE    2000  20100  
2                 2019-08-29  XXXXXXMFETF  XXXXXXOGEQQ   20000  20100  
3                 2019-08-28  XXXXXXMEMCC  XXXXXXCCFQM   10000  20100  
4                 2019-08-28  XXXXXXMFETF  XXXXXXGOPOE     500  20100  
...                      ...          ...          ...     ...    ...  
3729              2019-08-27  XXXXXXOQFTA  XXXXXXAOTCQ    3000  20000  
3730              2019-08-27  XXXXXXCTOET  XXXXXXFFAOC    2000  10000  
3731              2019-08-29  XXXXXXOCOFT  XXXXXXFETMT   10000  10000  
3732              2019-08-26  XXXXXXMCFQA  XXXXXXMFETF     100  10000  
3733              2019-08-28  XXXXXXGAGTO  XXXXXXACOMP     400  10000  

[3734 rows x 6 columns]

In [26]:
# those split transactions whose total amount is less than or equal to the limit of transaction
# are suspicious transactions

suspicious_round_split_transactions = merged[merged['Amount']<=merged['Limit']]
suspicious_round_split_transactions

KO ID / Type of Transaction  \
0     AAAORAUTPSCQXOGVNELKNCB187842400C / Money Tran...   
1     AAETIGMJYSSPYTYXPASJWSD580547108H / Money Tran...   
2     AAMIHDZNHEQEDPJNONUMGHB649383828V / Money Tran...   
3     AAPGYETNHRFYWNARXFDRXWJ710966481L / Money Tran...   
4     ABIURLGMYPUCODQTHLKKGOR654931422Z / Money Tran...   
...                                                 ...   
3729  ZVPETZNJGMLVCLFUWMXXCMB565230402U / Fund Transfer   
3730  FYKWLNFXAVGGSPCQVMYMFUG701624369F / AEPS ONUS ...   
3731  IVWTTRJEVPHZCLHUORTVKQR937821643B / AEPS ONUS ...   
3732  HQOEVKFLNTBGNCHLVWFOAMC261533838U / YONO WITHD...   
3733  UUZRCLIQCEKVTCBJHQJSJPS569050463H / AEPS OFFUS...   

     Transaction Date & Time From Account   To Account  Amount  Limit  
0                 2019-08-29  XXXXXXMFETF  XXXXXXEFMEG    3663  20100  
1                 2019-08-27  XXXXXXMFETF  XXXXXXTAGTE    2000  20100  
2                 2019-08-29  XXXXXXMFETF  XXXXXXOGEQQ   20000  20100  
3                 2019-08-28  XXXXXXMEMCC  XXXXXXCCFQM   10000  20100  
4                 2019-08-28  XXXXXXMFETF  XXXXXXGOPOE     500  20100  
...                      ...          ...          ...     ...    ...  
3729              2019-08-27  XXXXXXOQFTA  XXXXXXAOTCQ    3000  20000  
3730              2019-08-27  XXXXXXCTOET  XXXXXXFFAOC    2000  10000  
3731              2019-08-29  XXXXXXOCOFT  XXXXXXFETMT   10000  10000  
3732              2019-08-26  XXXXXXMCFQA  XXXXXXMFETF     100  10000  
3733              2019-08-28  XXXXXXGAGTO  XXXXXXACOMP     400  10000  

[3729 rows x 6 columns]

In [27]:
# creating an excel file of the suspicious round + split transactions

suspicious_round_split_transactions.to_excel(r'Suspicious Round + Split Transactions.xlsx',index = False)